mypy_django_plugin/transformers/managers.py

```diff
@@ -0,0 +1,183 @@
+        prepared_arguments.append(argument)
+    return_type = AnyType(TypeOfAny.unannotated)
+    return prepared_arguments, return_type
+
+
+class ManagerCallback(helpers.DynamicClassPluginCallback):
+
+    def copy_method_to_another_class(
+            self,
+            ctx: ClassDefContext,
+            self_type: Instance,
+            new_method_name: str,
+            method_node: FuncDef) -> None:
+        if method_node.type is None:
+            if not self.semanal_api.final_iteration:
+                self.semanal_api.defer()
+                return
+
+            arguments, return_type = build_unannotated_method_args(method_node)
+            add_method_to_class(
+                    ctx.api,
+                    ctx.cls,
+                    new_method_name,
+                    args=arguments,
+                    return_type=return_type,
+                    self_type=self_type)
+            return
+
+        method_type = method_node.type
+        if not isinstance(method_type, CallableType):
+            if not self.semanal_api.final_iteration:
+                self.semanal_api.defer()
+            return
+
+        arguments = []
+        bound_return_type = self.semanal_api.anal_type(
+                method_type.ret_type,
+                allow_placeholder=True)
+
+        assert bound_return_type is not None
+
+        if isinstance(bound_return_type, PlaceholderNode):
+            return
+
+        for arg_name, arg_type, original_argument in zip(
+                method_type.arg_names[1:],
+                method_type.arg_types[1:],
+                method_node.arguments[1:]):
+            bound_arg_type = self.semanal_api.anal_type(arg_type, allow_placeholder=True)
+            if bound_arg_type is None and not self.semanal_api.final_iteration:
+                self.semanal_api.defer()
+                return
+
+            assert bound_arg_type is not None
+
+            if isinstance(bound_arg_type, PlaceholderNode):
+                return
+
+            var = Var(
+                    name=original_argument.variable.name,
+                    type=arg_type)
+            var.line = original_argument.variable.line
+            var.column = original_argument.variable.column
+            argument = Argument(
+                    variable=var,
+                    type_annotation=bound_arg_type,
+                    initializer=original_argument.initializer,
+                    kind=original_argument.kind)
+            argument.set_line(original_argument)
+            arguments.append(argument)
+
+        add_method_to_class(
+                ctx.api,
+                ctx.cls,
+                new_method_name,
+                args=arguments,
+                return_type=bound_return_type,
+                self_type=self_type)
+
+
+class ManagerFromQuerySetCallback(ManagerCallback):
+    def create_new_dynamic_class(self) -> None:
+        callee = self.call_expr.callee
+
+        assert isinstance(callee, MemberExpr)
```

Comment:
I assume for all `get_dynamic_class_hook` invoked on methods, this will be a `MemberExpr`. Could you add a generic subclass of `DynamicClassPluginCallback` called `DynamicClassFromMethodCallback` with `callee` present as a class attribute and set in the `__call__`?